packages/mesos-client/src/observableFromXhrConnection.js

```diff
@@ -0,0 +1,30 @@
+import { ConnectionConstants, ConnectionManager } from "connection-manager";
+import Rx from "rxjs";
+
+export default function observableFromXhrConnection(connection) {
+  return Rx.Observable.create(function(observer) {
+    connection.addListener(
+      ConnectionConstants.CONNECTION_EVENT_PROGRESS,
+      function(target) {
+        observer.next(target.response);
+      }
+    );
+    connection.addListener(ConnectionConstants.CONNECTION_EVENT_ERROR, function(
+      error
+    ) {
+      observer.error(error);
+    });
+    connection.addListener(
+      ConnectionConstants.CONNECTION_EVENT_CLOSE,
+      function() {
+        observer.complete();
+      }
+    );
+
+    ConnectionManager.queue(connection);
+
+    return function() {
+      connection.close();
```

Comment:
Is there a chance the `connection` is already closed? Couldn't this function be invoked because `ConnectionConstants.CONNECTION_EVENT_CLOSE` was triggered?

Reply:
Yes, that can happen, but then `Observable` will be `completed`. This piece of logic is responsible for cleaning up resources after all Observers have unsubscribed from the Observable.

Reply:
@nLight let's name this method to it is easier to understand.